doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -1275,14 +1231,21 @@ constructions.
 
 .. _Fixpoint:
 
-.. cmd:: Fixpoint @ident @binders {? {struct @ident} } {? : @type } := @term
+.. cmd:: Fixpoint @fix_definition {* with @fix_definition }
```

Comment:
This includes the syntax for mutually recursive definitions but no corresponding explanation is given here, only in the `cmdv` below. What can we do about this? We should either, include the explanation on mutually recursive definitions sooner, or change this to `Fixpoint @fix_definition` and leave the full syntax for the variant below.

Reply:
Should this be as shown, or maybe `{? : type } {? := term }`?
![image](https://user-images.githubusercontent.com/1253341/74711222-50e21100-51d8-11ea-8574-4a40dc2bd4ce.png)

Reply:
I've not tried to clean up all the cmdv's--seemed like you might have to clean up a lot of text.  And understand it in some depth.

This one is partial:

![image](https://user-images.githubusercontent.com/1253341/74711455-dc5ba200-51d8-11ea-9cff-922b36bd24ae.png)

- Missing part of fixannot
- := term is optional

Reply:
Also this seems weird: "This command allows defining functions by pattern matching over inductive objects using a fixed point construction." for which it seems a `match` is required, but the syntax doesn't cover this (presumably it's a restriction on `@term`).  Not sure what to say here.

Reply:
- Yes, I was suggesting the use of `{? : type }`.

- It would be great if we could take a decision on whether to remove or to clean this particular `cmdv`.

- There is no syntactic restriction and indeed, you can use `Fixpoint` without a `match` like so: `Fixpoint foo (n : nat) := 0.` (this is not actually useful because there is no recursive call). The verification that the recursive calls are done on structurally smaller arguments is enough to generally imply that `term` will contain a `match` construct.

Reply:
No syntactic restriction, nor (based on your example) is there any semantic restriction requiring `match`, right?

Reply:
Right. It's just that you need a `match` to write any meaningful recursive function.